### DISCUSSION TITLE
Replace Template Manager

### DISCUSSION THREAD

**Discussion by ns8482e** [OrchardCMS/OrchardCore#7001] [General]
I am working on creating a module that will replace the exiting `Templates` and `AdminTemplate` feature.   I can not replace TemplateManager in `services` using Dependency as it's not interface based. So Currently I have to manually disable `Templates` and `AdminTemplate` before enabling my module.

Is it possible to achieve in C# code that when user enables my module, it should disable other modules like `Templates` and `AdminTemplate`? 

Thank You

**Comment by Skrypt** [comment]
From your startup.cs class of your module you can unregister a service simply.

```
services.AddScoped<IService>();

var serviceDescriptor = services.FirstOrDefault(descriptor => descriptor.ServiceType == typeof(IService));
services.Remove(serviceDescriptor);
```

Though you must make sure that your module is registered after the Templates and AdminTemplates ones.

If you look at the StartupBase class you will see ConfigureOrder which can be set like this : 

```
public override int ConfigureOrder => 2000;
```

The higher the value ; the later it gets registered.

**Reply by ns8482e** [reply]
My first try was that - to have my module has dependency on  `OrchardCore.Templates` and `OrchardCore.AdminTemplates` and then I could replace `AdminTemplatesManager` but it is `services.AddScoped<AdminTemplatesManager>();` added as concrete dependency, I can't remove it from my module without adding reference to `OrchardCore.Templates` module. 

Is it okay to have one module to have reference to another module?
